pkg/openSearch/openSearchQuery/compareHandler.go

```diff
@@ -269,3 +271,27 @@ func getStringRange(fieldName string, rating string, querySettings *QuerySetting
 	}
 	return RatingRange{}
 }
+
+// HandleCompareOperatorOnDay creates an opensearch range query for a given date field.
+// It ensures the field value is a valid RFC3339Nano date string and constructs a query
+// that matches all timestamps within the specified day (from 00:00:00 to 23:59:59 UTC).
+// it returns a MatchNone query if the operation fails
+func HandleCompareOperatorOnDay(fieldName string, fieldKeys []string, fieldValue any, querySettings *QuerySettings) esquery.Mappable {
+	stringValue, ok := fieldValue.(string)
+	if !ok {
+		log.Error().Msgf("fieldValue is not a string: %v", fieldValue)
+		return esquery.MatchNone()
+	}
+
+	date, err := time.Parse(time.RFC3339Nano, stringValue)
+	if err != nil {
+		log.Error().Err(err).Msgf("failed to parse date string: %s", stringValue)
```

Comment:
```suggestion
		log.Err(err).Msgf("failed to parse date string: %s", stringValue)
```
log.Err(err) already sends error level log

Reply:
Thanks. 
Changed